src/vm/ecalllist.h

```diff
@@ -1603,10 +1603,10 @@ FCFuncStart(gInteropMarshalFuncs)
     FCFuncElement("DestroyStructure", MarshalNative::DestroyStructure)
     FCFuncElement("UnsafeAddrOfPinnedArrayElement", MarshalNative::FCUnsafeAddrOfPinnedArrayElement)
     FCFuncElement("GetExceptionCode", ExceptionNative::GetExceptionCode)
+    QCFuncElement("GetHINSTANCE", COMModule::GetHINSTANCE)
```

Comment:
Yes mono returns -1. I have change it to ensure we return -1 for non-windows platform as well. I have updated the PR and squashed the changes.